cypress/integration/console/devices/edit.spec.js

```diff
@@ -0,0 +1,165 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Device general settings', () => {
+  const appId = 'end-device-edit-test-application'
+  const application = { ids: { application_id: appId } }
+  const userId = 'end-device-edit-test-user'
+  const user = {
+    ids: { user_id: userId },
+    primary_email_address: 'end-device-edit-test-user@example.com',
+    password: 'ABCDefg123!',
+    password_confirm: 'ABCDefg123!',
+  }
+  const endDeviceId = 'end-device-edit-test'
+  const endDevice = {
+    application_server_address: 'localhost',
+    ids: {
+      device_id: endDeviceId,
+      dev_eui: '0000000000000001',
+      join_eui: '0000000000000000',
+    },
+    name: 'End Device Test Name',
+    description: 'End Device Test Description',
+    join_server_address: 'localhost',
+    network_server_address: 'localhost',
+  }
+  const endDeviceFieldMask = {
+    paths: [
+      'join_server_address',
+      'network_server_address',
+      'application_server_address',
+      'ids.dev_eui',
+      'ids.join_eui',
+      'name',
+      'description',
+    ],
+  }
+  const endDeviceRequestBody = {
+    end_device: endDevice,
+    field_mask: endDeviceFieldMask,
+  }
+
+  before(() => {
+    cy.dropAndSeedDatabase()
+    cy.createUser(user)
+    cy.createApplication(application, userId)
+    cy.createEndDevice(appId, endDeviceRequestBody)
```

Comment:
Please use `createMockDeviceAllComponents()` here to make the test more realistic. Also allows you to get rid of the definitions above.